### ISSUE TITLE
How to load policy.xml on Azure Windows?

### ISSUE THREAD

**Issue by msangapu** [ImageMagick/ImageMagick#1379]
### Description
ImageMagick does not seem to be honoring the policy.xml file. The file is located in a directory that is in the MAGIC_CONFIGURE_PATH, however, it is not being honored. What is the best way to load policy.xml on IIS/Windows?

### Steps to Reproduce
Easy to reproduce on Azure Windows. Policy.xml not being honored.

### System Configuration
Azure Windows (IIS)

**Comment by dlemstra** [comment]
Is `MAGIC_CONFIGURE_PATH` a typo and did you mean to type `MAGICK_CONFIGURE_PATH` instead? And how ar you setting `MAGICK_CONFIGURE_PATH`?

**Comment by msangapu** [comment]
Yes, it's a typo. I've since updated it. I'm adding it through applicationHost.xdt (IIS Transform). It's how I even got ImageMagick to work in the first place. I can output the value in the test page if needed. 

Azure>Kudu showing the environment variables and their values:
MAGICK_CONFIGURE_PATH = d:\home\site\imagick\
MAGICK_HOME = d:\home\site\imagick\

**Comment by msangapu** [comment]
I've updated the test page to print $_SERVER['MAGICK_CONFIGURE_PATH'] value in the PDF

**Comment by msangapu** [comment]
I've updated it to use getenv instead. Same behavior.

**Comment by dlemstra** [comment]
What you could do is enable debug logging but I have no idea how you would need to do that in PHP.

**Comment by msangapu** [comment]
Does PHP-ImageMagick have parameters for debug logs? I can enable PHP Error Logging (set to E_ALL).

I believe PHP is allowed to read policy.xml -- the image formats are located in a separate path which I'm also referencing through an environment variable and can be verified through phpinfo(); so I can only assume policy.xml will go by the same rules. It would be a "nice to have" to have policy.xml details listed in phpinfo();

Upgrading to a new version of PHP-ImageMagick on Windows is not easy. I had to do a lot of testing until I got the right version matches between the PHP Extension and ImageMagick DLLs. I documented the steps https://blogs.msdn.microsoft.com/azureossds/2015/12/07/php-imagemagick-on-azure-web-apps/ , after doing a lot of tinkering on my machine. If you have a short-cut or know which versions are compatible, I can test it out.

But my instinct tells me this is related to PHP honoring policy.xml. If there is some debugging I can enable then I can do some testing.

**Comment by dlemstra** [comment]
We did not write and do not maintain PHP-ImageMagick. I wonder if ImageMagick gets loaded before you are able to set the `MAGICK_CONFIGURE_PATH`.

It would probably be a good idea to add some more information to `phpinfo()` or create a `magickinfo()` instead but you would need to ask the PHP-ImageMagick team that.

There is support for debug logging and you can change that in the `log.xml` file but you will probably run into the same issue again as now with the `policy.xml` file. You need to modify the file at the location where ImageMagick is installed instead.

**Comment by msangapu** [comment]
I've reached out to the authors of the PHP Extension and included a link to this thread. I'm hoping one of them will able to assist me further. I hope it's okay to keep this issue open until I hear back from one of the authors. Thank you for your assistance and contributions to Open Source.
